docker-app/qfieldcloud/core/middleware/test.py

```diff
@@ -6,8 +8,7 @@ class TestMiddleware:
     def __init__(self, get_response):
         self.get_response = get_response
 
-        # TODO check if in test mode, if not raise exception
-        if False:
+        if os.environ.get("ENVIRONMENT") != "test":
             raise MiddlewareNotUsed()
```

Comment:
Not sure if there's any particular logic about this in Django ? Is this catched by Django to disable the middleware ? Otherwise this will break production.

An alternative maybe more readable implementation would be to disable it in settings.py like this

```python
# settings.py
MIDDLEWARE = [
   # ...
]
if os.environ.get("ENVIRONMENT") == "test":
    MIDDLEWARE.append("qfieldcloud.core.middleware.test.TestMiddleware")
```

Reply:
I was following the docs:

https://docs.djangoproject.com/en/3.2/topics/http/middleware/#marking-middleware-as-unused

Reply:
Ah ok fine then :-)